ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/HiveMetaStoreAuthorizer.java

```diff
@@ -213,15 +214,46 @@ public final List<String> filterCatalogs(List<String> catalogs) throws MetaExcep
 
   @Override
   @Deprecated
-  public List<TableMeta> filterTableMetas(String catName, String dbName,List<TableMeta> tableMetas)
+  public List<TableMeta> filterTableMetas(String catName, String dbName, List<TableMeta> tableMetas)
       throws MetaException {
-    return filterTableMetas(tableMetas);
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("==> HiveMetaStoreAuthorizer.filterTableMetas()");
+    }
+    List<String> tableNames = new ArrayList<>();
+    List<TableMeta> filteredTableMetas = new ArrayList<>();
+    if (tableMetas != null) {
+      for (TableMeta tableMeta : tableMetas) {
+        tableNames.add(tableMeta.getTableName());
+      }
+      TableFilterContext     tableFilterContext     = new TableFilterContext(dbName, tableNames);
+      HiveMetaStoreAuthzInfo hiveMetaStoreAuthzInfo = tableFilterContext.getAuthzContext();
+      final List<String>  filteredTableNames = filterTableNames(hiveMetaStoreAuthzInfo, dbName, tableNames);
+      if (CollectionUtils.isEmpty(filteredTableNames)) {
+        filteredTableMetas = Collections.emptyList();
+        if (LOG.isInfoEnabled()) {
+          LOG.info("<== HiveMetaStoreAuthorizer.filterTableMetas() : returning empty set");
+        }
+      } else {
+        if (LOG.isDebugEnabled()) {
```

Comment:
No need to check LOG.isDebugEnabled(), since LOG.debug will only output, when debug level logging is enabled.